File: app/src/main/java/ru/ls/donkitchen/db/table/Category.java
package ru.ls.donkitchen.db.table;

import android.provider.BaseColumns;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

/**
 * @author Lord (Kuleshov M.V.)
 * @since 11.11.15
 */
@DatabaseTable(tableName = "category")
public class Category implements BaseColumns {
	public static final String NAME = "name";
	public static final String IMAGE_LINK = "image_link";
	public static final String RECEIPT_COUNT = "receipt_count";
	public static final String PRIORITY = "priority";

	@DatabaseField(id = true, canBeNull = false, columnName = _ID)
	public int id;
	@DatabaseField(canBeNull = false, columnName = NAME)
	public String name;
	@DatabaseField(canBeNull = false, columnName = IMAGE_LINK)
	public String imageLink;
	@DatabaseField(columnName = RECEIPT_COUNT)
	public int receiptCount;
	@DatabaseField(columnName = PRIORITY)
	public int priority;

	@Override
	public String toString() {
		return "Category{" +
				"id=" + id +
				", name='" + name + '\'' +
				", imageLink='" + imageLink + '\'' +
				", receiptCount=" + receiptCount +
				", priority=" + priority +
				'}';
	}
}

File: data/src/main/kotlin/ru/ls/donkitchen/data/db/table/Category.kt
package ru.ls.donkitchen.data.db.table

import android.provider.BaseColumns
import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable

/**
 *
 *
 * @author Lord (Kuleshov M.V.)
 * @since 16.10.16
 */
@DatabaseTable(tableName = "category")
class Category : BaseColumns {
    @DatabaseField(id = true, canBeNull = false, columnName = BaseColumns._ID)
    var id: Int = 0
    @DatabaseField(canBeNull = false, columnName = NAME)
    var name: String? = null
    @DatabaseField(canBeNull = false, columnName = IMAGE_LINK)
    var imageLink: String? = null
    @DatabaseField(columnName = RECEIPT_COUNT)
    var receiptCount: Int = 0
    @DatabaseField(columnName = PRIORITY)
    var priority: Int = 0

    companion object {
        const val NAME = "name"
        const val IMAGE_LINK = "image_link"
        const val RECEIPT_COUNT = "receipt_count"
        const val PRIORITY = "priority"
    }

    override fun toString(): String {
        return "Category(id=$id, name=$name, imageLink=$imageLink, receiptCount=$receiptCount, priority=$priority)"
    }
}
